test/transaction.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file transaction.cpp
+ * @author Dmitrii Khokhlov <winsvega@mail.ru>
+ * @date 2014
+ * Transaaction test functions.
+ */
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/test/unit_test.hpp>
+#include "JsonSpiritHeaders.h"
+#include <libdevcore/CommonIO.h>
+#include <libethereum/BlockChain.h>
+#include <libethereum/State.h>
+#include <libethereum/ExtVM.h>
+#include <libethereum/Defaults.h>
+#include <libevm/VM.h>
+#include "TestHelper.h"
+
+using namespace std;
+using namespace json_spirit;
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::eth;
+
+namespace dev {  namespace test {
+
+Transaction createTransactionFromFields(mObject& _tObj)
+{
+	BOOST_REQUIRE(_tObj.count("data") > 0);
+	BOOST_REQUIRE(_tObj.count("value") > 0);
+	BOOST_REQUIRE(_tObj.count("gasPrice") > 0);
+	BOOST_REQUIRE(_tObj.count("gasLimit") > 0);
+	BOOST_REQUIRE(_tObj.count("nonce")> 0);
+	BOOST_REQUIRE(_tObj.count("to") > 0);
+
+	BOOST_REQUIRE(_tObj.count("v") > 0);
+	BOOST_REQUIRE(_tObj.count("r") > 0);
+	BOOST_REQUIRE(_tObj.count("s") > 0);
+
+	//Construct Rlp of the given transaction
+	RLPStream rlpStream;
+	rlpStream.appendList(9);
+	rlpStream << toInt(_tObj["nonce"]) << toInt(_tObj["gasPrice"]) << toInt(_tObj["gasLimit"]);
```

Comment:
toInt is making a down cast. We should only use strings in the json file and cast to bigint.